services/oauth/v2/destination_info_test.go

```diff
@@ -109,6 +112,65 @@ var isOAuthDestTestCases = []destInfoTestCase{
 			isOAuth: false,
 		},
 	},
+	{
+		description:    "success scenario for a oauth destination where account is present",
+		flow:           common.RudderFlowDelivery,
+		inputDefConfig: map[string]interface{}{},
+		account: &backendconfig.AccountWithDefinition{
+			AccountDefinition: backendconfig.AccountDefinition{
+				Config: map[string]interface{}{
+					"refreshOAuthToken": true,
+				},
+			},
+		},
+		expected: isOAuthResult{
+			isOAuth: true,
+		},
+	},
+	{
+		description:    "success scenario for a non-oauth destination where account is not present",
+		flow:           common.RudderFlowDelivery,
+		inputDefConfig: map[string]interface{}{},
+		account:        nil,
+		expected: isOAuthResult{
+			isOAuth: false,
+		},
+	},
+	{
+		description:    "success scenario for a non-oauth destination where account is present",
+		flow:           common.RudderFlowDelivery,
+		inputDefConfig: map[string]interface{}{},
+		account: &backendconfig.AccountWithDefinition{
+			AccountDefinition: backendconfig.AccountDefinition{
+				Config: map[string]interface{}{
+					"refreshOAuthToken": false,
+				},
+			},
+		},
+		expected: isOAuthResult{
+			isOAuth: false,
+		},
+	},
+	{
+		description:    "failure scenario for a oauth destination where account is present and account config is wrong",
+		flow:           common.RudderFlowDelivery,
+		inputDefConfig: map[string]interface{}{},
+		account: &backendconfig.AccountWithDefinition{
+			AccountDefinition: backendconfig.AccountDefinition{
+				Config: map[string]interface{}{
+					"refreshToken": false,
+				},
+			},
+		},
+		expected: isOAuthResult{
+			isOAuth: false,
+			err: &misc.MapLookupError{
```

Comment:
should we throw an error here ? Config can have any other valid values (in future) for non-oauth destinations where `refreshOAuthToken` flag need not be

Reply:
yes